snapcraft/plugins/colcon.py

```diff
@@ -632,6 +632,9 @@ def _build_colcon_packages(self):
         # Specify that the packages should be installed into the overlay
         colconcmd.extend(["--install-base", self._ros_overlay])
 
+        # Specify the number of workers
+        colconcmd.append("--parallel-workers {}".format(self.parallel_build_count))
```

Comment:
This is combining the flag and the parameter as a single arg.

Simply switch:
`colconcmd.append("--parallel-workers {}".format(self.parallel_build_count))`

with:
`colconcmd.extend(["--parallel-workers", self.parallel_build_count])`

Now it's appending the flag, then appending the count.

Reply:
Alternatively, you could have done:
`colconcmd.append("--parallel-workers={}".format(self.parallel_build_count))`

Reply:
Plain curiosity, why is it actually a problem ? The generated command, as printed in CI terminal looks fine.

Any preference on which solution to use ? I'd go with the second one.

Reply:
It's due to argument parsing and the log isn't showing the quotes around the arguments.

It's the equivalent of you running:
`colcon build "--parallel-workers 1"`

The application (colcon) parses the string `--parallel-workers 1` as one argument, not as two.

Reply:
https://github.com/snapcore/snapcraft/blob/master/snapcraft/internal/errors.py#L577 is where it loses the quotes.

If we swapped out:
`command = " ".join(command)`

for:
`command = " ".join([shlex.quote(c) for c in command])`

We'd see the correct output.

Reply:
We use `shell=False` in our eventual `Popen` calls (default behavior) to avoid injection (https://docs.python.org/2/library/subprocess.html#frequently-used-arguments) so your arguments need to be sliced up.

Reply:
Ok I see, thanks !